New reporters and new reporters variations

quevon24 opened the pull request:
**_NOTE:_** the changes were divided into a PR for reporters and another for laws.

This is the list of new reporters:

### **Reporters:**

1. Abbott's New Jersey Law Reports
2. American Bankruptcy Law Journal
3. American Bar Association Journal (ABA)
4. American Lawyer
5. Boston University Law Review
6. Bradford's Surrogate Reports [NY]
7. Business Lawyer
8. California Workers' Compensation Cases
9. Chief Counsel Notice LEXIS cite
10. Civil Aeronautics Authority Reports
11. Colorado Journal, Colorado Appellate Reports
12. Constitution of The United States of America, USCS
13. Court Decisions Relating National Labor Relations Act
14. Current Court Decisions
15. Customs Appeals Decisions
16. Customs Decisions
17. Detroit Legal News
18. Employee & Labor Relations Law
19. Federal Rules of Evidence, USCS
20. Field Service Advice Memoranda
21. Field Service Advice Memoranda LEXIS cite
22. Florida (FL) Law Weekly Federal, Court of Appeals
23. Florida (FL) Law Weekly Federal, District Court
24. Florida (FL) Law Weekly, Circuit Court
25. Florida (FL) Law Weekly, District Courts of Appeal
26. Fourth Circuit and District of Columbia (DC) Bankruptcy Court Reporter
27. General Counsel Memoranda
28. Internal Revenue Bulletin LEXIS cite
29. LexisNexis Illinois Supreme Court
30. Municipal Law Reporter
31. Northwestern University Law Review
32. Notre Dame Lawyer
33. Ohio (OH) Pending Opinions Reporter
34. Thompson National Bank Cases
35. Treasury Decisions
36. Treasury Decisions Under Customs & Other Laws
37. United States Court of Appeals for the Federal Circuit
38. United States Court of International Trade Slip Opinion Citation
39. United States Granted Patent Number
40. United States Treaties and Other International Agreements (TIAS)

This PR also contains many reporters variations, new regex patterns: full_cite_cch_suffix, volume_with_letter and updates in regex patterns

quevon24 committed:
feat(variations): New variations for reporters

quevon24 committed:
feat(reporters): New reporters and reporters variations

Cal. WCC, CJ C.A.R., Fla. L. Weekly Fed. D, Thomp.Nat.Bank Cas.,
Va.Sup.Ct.Rep.

quevon24 committed:
feat(variations): Reporters variations

Variations, modify regex formats and new regex format
'full_cite_cch_suffix'

quevon24 committed:
feat(reporters): New reporter and reporters variations

Abbotts, Fla. L. Weekly Fed. C, Fourth Cir. & D.C. Bankr., Current Ct.Dec.

quevon24 committed:
feat(reporters): New reporters variations

quevon24 committed:
feat(reporters): New reporters and more reporters variations

Fla. L. Weekly D, Mun.L.R., Detroit Leg.N.

quevon24 committed:
feat(reporters): Reporters variations from lexis data

quevon24 committed:
feat(reporters): New reporters and variations from lexis data

Ohio Pending Op. Rep., C.A.D.

quevon24 committed:
feat(reporters): New reporters and variations from lexis data

Fla. L. Weekly C, Ct. Dec. Relating N.L.R.A., U.S.C., C.F.R., FLA. STAT.

quevon24 committed:
feat(reporters): Reporters variations and new reporters

WIS. STAT., VA. CODE ANN., WASH. REV. CODE, WYO. STAT. ANN.

quevon24 committed:
feat(reporters): new reporters and reporters variations

W. VA. CODE, Williams & Meyers, Oil and Gas Law, UTAH CODE ANN., U.S.T.,
USSG, T.D., S.C. CODE ANN., Rev. Proc., Rev. Rul., OR. REV. STAT., OHIO
REV. CODE ANN., O.C.G.A., N.Y. Estates, Powers Trusts Law, N.Y. Dom.
Rel. Law, N.Y. Debt. Cred. Law

quevon24 committed:
feat(reporters): new reporters and reporters variations

Fed. Tax Ct. R., GCM, Ind. Code, Iowa Code, Internal Revenue Manual
Section, IRC, IRM, Kan. Stat. Ann., Ky. Const., Ky. Rev. Stat. Ann., La.
R.S., La. C.C., MD. CODE ANN., Mich. Comp. Laws, Miss. Code Ann., Mont.
Code Ann.

quevon24 committed:
feat(reporters): new reporters and reporters variations

Fed. R. Evid, Fed. R. Crim. P., Fed. R. Civ. P., Fed. R. Bankr. P., Fed. R. App. P., Exec. Order, Empl. & L. Rel. Law, D. Minn. LR, D. Nev. LR, Del. Code Ann., D.C. Code

quevon24 committed:
feat(reporters): new reporters and reporters variations

Ala. Code, Alaska Stat., Appendix U.S.C., Ariz. Rev. Stat., Ark. Code Ann., Cal. Bus. & Prof. Code, Cal. Civ. Code, Cal. Civ. Proc. Code, Cal. Com. Code

quevon24 committed:
feat(reporters): new reporters and reporters variations

Cal. Corp. Code, Cal. Educ. Code, Cal. Elec. Code, Cal. Evid. Code, Cal. Fam. Code, Cal. Fin. Code, Cal. Food & Agric. Code, Cal. Gov't Code, Cal. Health & Safety Code, Cal. Ins. Code, Cal. Lab. Code, Cal. Penal Code, Cal. Prob. Code, Cal. Pub. Res. Code, Cal. Rev. & Tax. Code, Cal. Unemp. Ins. Code, Cal. Veh. Code

quevon24 committed:
feat(reporters): new reporters and reporters variations

Cal. Welf. & Inst. Code, CCN LEXIS, Haw. Rev. Stat., Hous. Bus. & Tax L.J., Idaho Code, Ill. Comp. Stat., Ill. Sup. Ct. INTERNAL REVENUE MANUAL (CCH), IRB LEXIS, Ill. L. Rev., Judgments, Judgments Second, Mass. Ann. Laws, Mo. Rev. Stat., M.R.S, N.C. Gen. Stat., N.D. Ala. Rules, N.D. Cent. Code, N.D. Ga. LCrR, Neb. Rev. Stat., N.H. Rev. Stat. Ann., N.J. Stat. Ann., N.M. Stat. Ann., NRS, N.Y. Banking Law, N.Y. Bus. Corp. Law, N.Y. City Admin. Code, N.Y. City Charter, N.Y. Civ. Rights Law, N.Y. Civ. Serv. Law, N.Y. C.P.L.R., N.Y. Crim. Proc. Law, N.Y. Educ. Law, N.Y. Envtl. Conserv. Law, N.Y. Fam. Ct. Act, N.Y. Tax Law

quevon24 committed:
feat(reporters): new reporters

Pa.C.S., PLR, Property Third Servitudes, Prop. Treas. Reg., Pat. No, P.S., Restatement of Property, Restitution, Restatement of Security, Agency, R.I. Gen. Laws, S.D. Codified Laws, S.D. Ind., SLIP OP., Sup. Ct. R, Tenn. Code Ann., Tex. Bus. & Com. Code, Tex. Bus. Corp. Act, Tex. Bus. Orgs. Code, Tex. Civ. Prac. & Rem. Code, Tex. Const., Tex. Educ. Code, Tex. Fam. Code,Tex. Gov't Code, Tex. Ins. Code

quevon24 committed:
feat(reporters): new reporters

Tex. Penal Code, Tex. Prob. Code, Tex. Prop. Code, Tex. Rev. Civ. Stat., Tex. Tax Code, Torts Second, Trusts Second, Trusts Third, Treas. Dec., Treas. Reg., U.S. CONST., V.I.C., Vt. Stat. Ann, Wash. Rev. Code, N.Y. Gen. Bus. Law, N.Y. Gen. City Law, N.Y. Gen. Constr. Law, N.Y. Gen. Mun. Law, N.Y. Gen. Oblig. Law, N.Y. High. Law, N.Y. Ins. Law, N.Y. Jud. Law, N.Y. Lien Law, N.Y. Limited Liab. Co. Law, N.Y. Mental Hyg. Law, N.Y. Not-for-Profit Corp. Law, N.Y. Partnership Law, N.Y. Penal Law, N.Y. Pers. Prop. Law, N.Y. Priv. Hous. Fin. Law

quevon24 committed:
feat(reporters): new reporters

N.Y. Real Prop. Law, N.Y. Real Prop. Acts. Law, N.Y. Real Prop. Tax Law, N.Y. Relig. Corp. Law, N.Y. Retire. & Soc. Sec. Law, N.Y. Soc. Serv. Law, N.Y. Surr. Ct. Proc. Act, N.Y. Veh. & Traf. Law, Notre Dame Lawyer, Nw. U.L. Rev, Okla. Stat.

quevon24 committed:
feat(reporters): new reporters and reporters variations

A.B.A.J., Agency Second, Agency Third, Ala. Code, Announcement, Bankr. R., B.U.L. Rev., Bradf. Surr., Bus. Law., Colo. Rev. Stat., Conflict of Law Second, Conn. Gen. Stat., Contracts Second, Cust. Dec., Foreign Relations Law Third, Foreign Relations Law Second, FSA, FSA LEXIS, Fed. Cir. (T)

quevon24 committed:
fix(reporters): fix Internal Revenue Manual (CCH) reporter

quevon24 committed:
fix(reporters): update Am. Jur. and new reporters

Am. Bank. L.J., Am. Law., C.A.B.

quevon24 commented:
New reporters and new variations are ready @flooie 🤞

mlissner commented:
Wow wow wow. Awesome.

quevon24 committed:
Merge remote-tracking branch 'origin/main' into west-citations-variations

flooie commented on reporters_db/data/regexes.json:
```diff
@@ -78,6 +78,8 @@
         "with_alpha_suffix#": "Volume number that may have 'A' appended, like '1A'",
         "with_digit_suffix": "(?P<volume>\\d{1,4}(?:-\\d+)?)",
         "with_digit_suffix#": "Volume number that may have digits appended, like '123-4'",
+        "with_letter": "(?P<volume>\\d+[a-zA-Z])",
+        "with_letter#": "Volume number followed by a letter",
```
I'm a little worried about typos getting sucked into these by allowing any Letter.

quevon24 replied:
I updated the regex with the letters i found in the citations

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -1320,6 +1389,27 @@
+            "name": "Announcement",
+            "variations": {}
+        }
+    ],
```
Where is this from?  When ? This is such an odd looking reporter name to me

quevon24 replied:
Some of the reporters came from lexis nexis shepard's citation formats, you can take a look here: https://www.lexisnexis.com/pdf/lexis-advance/Shepards-Citation-Formats.pdf

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -4588,14 +4760,15 @@
+                "CAL. PENAL CODE 532A"
             ],
             "mlz_jurisdiction": [],
             "name": "California Statutes: Deering's Cal. Codes Ann. - Penal Code",
```
I assume this also should be in laws.json.

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -4659,14 +4832,15 @@
-                "Cal. Rev. & Tax. Code § 19900"
+                "Cal. Rev. & Tax. Code § 19900",
+                "CAL. REV. TAX. CODE 23701D"
             ],
```
Same here.  These are laws ... right?

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -5757,6 +5931,29 @@
+                "COLO. REV. STAT.": "Colo. Rev. Stat."
+            }
+        }
+    ],
```
Same.

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -5934,6 +6153,30 @@
+                "CONN. GEN. STAT.": "Conn. Gen. Stat."
+            }
+        }
+    ],
```
Laws.json

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -8967,6 +9298,26 @@
+            "name": "United States Court of Appeals for the Federal Circuit",
+            "variations": {}
+        }
+    ],
```
What is the T for?  This is a weird format.

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -9798,6 +10149,42 @@
+                "FOREIGN RELATIONS LAW THIRD": "Foreign Relations Law Third"
+            }
+        }
+    ],
```
This should be the same reporter via editions?  
idk.

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -17766,7 +18161,8 @@
-                "N.J. STAT. ANN. 12A:7-204"
+                "N.J. STAT. ANN. 12A:7-204",
+                "N.J. STAT. ANN. 17:9A-19"
             ],
```
I assume these were added before we generated the laws.json...

flooie commented on reporters_db/data/reporters.json:
```diff
@@ -27302,7 +27702,7 @@
             ],
             "mlz_jurisdiction": [],
-            "name": "U.S. Constitution",
+            "name": "Constitution of The United States of America, USCS",
```
This seems odd... what is the USCS stand for?  Also probably meant for the laws.json

quevon24 replied:
I think it stands for "United States Code Service"

quevon24 committed:
feat(laws): Move reporters to laws, new in laws.json, update law section regex

Ala. Code, Alaska Stat., Cal. Civ. Code, Cal. Civ. Proc. Code, Cal. Com.
Code, Cal. Corp. Code, Cal. Educ. Code, Cal. Elec. Code, Cal. Evid.
Code, Cal. Fam. Code, Cal. Fin. Code, Cal. Food & Agric. Code, Cal.
Gov't Code, Cal. Health & Safety Code, Cal. Ins. Code, Cal. Lab. Code,
Cal. Penal Code, Cal. Prob. Code, Cal. Pub. Res. Code, Cal. Rev. & Tax.
Code, Cal. Unemp. Ins. Code, Cal. Veh. Code, Cal. Welf. & Inst. Code

quevon24 committed:
feat(laws): Move reporters to laws, update laws

Colo. Rev. Stat., Conn. Gen. Stat., Fla. Stat., Haw. Rev. Stat., Idaho Code, Ill. Comp. Stat., Kan. Stat. Ann., Ky. Rev. Stat. Ann., La. R.S., Me. Rev. Stat., Minn. Stat., Mo. Rev. Stat., N.C. Gen. Stat, N.D. Cent. Code, N.H. Rev. Stat. Ann., N.J. Stat. Ann., N.M. Stat. Ann., N.Y. Banking Law, N.Y. Bus. Corp. Law, N.Y. C.P.L.R., N.Y. Civ. Rights Law, N.Y. Civ. Serv. Law, N.Y. Crim. Proc. Law, N.Y. Educ. Law, N.Y. Envtl. Conserv. Law, N.Y. Fam. Ct. Act

quevon24 committed:
feat(laws): Move reporters to laws

N.Y. Gen. Bus. Law, N.Y. Gen. City Law, N.Y. Gen. Constr. Law, N.Y. Gen. Mun. Law, N.Y. Gen. Oblig. Law, N.Y. High. Law, N.Y. Ins. Law, N.Y. Jud. Law, N.Y. Lien Law, N.Y. Limited Liab. Co. Law, N.Y. Mental Hyg. Law, N.Y. Not-for-Profit Corp. Law, N.Y. Partnership Law, N.Y. Penal Law, N.Y. Pers. Prop. Law, N.Y. Priv. Hous. Fin. Law, N.Y. Real Prop. Acts. Law, N.Y. Real Prop. Law, N.Y. Real Prop. Tax Law, N.Y. Relig. Corp. Law, N.Y. Retire. & Soc. Sec. Law, N.Y. Soc. Serv. Law, N.Y. Surr. Ct. Proc. Act, N.Y. Tax Law, N.Y. Veh. & Traf. Law

quevon24 committed:
feat(laws): Move reporters to laws

NRS, Neb. Rev. Stat., Or. Rev. Stat., Okla. Stat., P.S., Pa.C.S., R.I. Gen. Laws, S.D. Codified Laws, Tenn. Code Ann., Tex. Bus. & Com. Code, Tex. Bus. & Com. Code, Tex. Bus. Corp. Act, Tex. Bus. Orgs. Code, Tex. Civ. Prac. & Rem. Code, Tex. Educ. Code, Tex. Fam. Code, Tex. Gov't Code, Tex. Ins. Code, Tex. Penal Code, Tex. Prob. Code, Tex. Prop. Code, Tex. Rev. Civ. Stat., Tex. Tax Code

quevon24 committed:
feat(laws): Move reporters to laws

Vt. Stat. Ann., Wis. Stat., WYO. STAT. ANN., Wash. Rev. Code, V.I. Code Ann.

quevon24 committed:
feat(laws): Move reporters to laws

Ark. Code Ann., WASH. REV. CODE, W. VA. CODE, VA. CODE ANN., UTAH CODE ANN., U.S.C., S.C. CODE ANN., OHIO REV. CODE ANN., MONT. CODE ANN., MD. CODE ANN., IOWA CODE, IND. CODE, C.F.R., D.C. Code, O.C.G.A., Cal Bus & Prof Code, IRC, La. C.C., N.Y. City Admin. Code, N.Y. City Charter

quevon24 committed:
refactor(reporters): Keep only reporters, remove new laws

new reporter variations, new reporters: Harv. L. Rev., Colum. L. Rev., Tax Law.

quevon24 committed:
feat(regexes): Update page_with_letter regex

quevon24 committed:
fix(reporter): Correct reporter format

quevon24 commented:
I've already reviewed all the changes in this PR, some of the reporters are already in reporters-db but there are several that we don't have along with some special examples, I will reopen this PR to work on this later.
